core/src/test/java/com/amazon/randomcutforest/RandomCutForestBuilderTest.java

```diff
@@ -73,8 +73,8 @@ public void testDefaultForestWithDimensionArgument() {
         assertEquals(256 / DEFAULT_OUTPUT_AFTER_FRACTION, f.getOutputAfter());
         assertFalse(f.storeSequenceIndexesEnabled());
         assertFalse(f.centerOfMassEnabled());
-        assertTrue(f.parallelExecutionEnabled());
-        assertEquals(Runtime.getRuntime().availableProcessors() - 1, f.getThreadPoolSize());
```

Comment:
Can we add a new test to verify that thread pool size is being set correctly when parallel execution is set to true but no explicit value is given for thread pool size?

Reply:
yes, i just added back this test case named testForestBuilderWithDefaultParallelExecutionThreadPoolSize .